hathor/nanocontracts/storage/node_nc_type.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2025 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from typing_extensions import override
+
+from hathor.nanocontracts.nc_types import (
+    BytesLikeNCType,
+    BytesNCType,
+    DictNCType,
+    NCType,
+    OptionalNCType,
+    VarUint32NCType,
+)
+from hathor.serialization import Deserializer, Serializer
+
+if TYPE_CHECKING:
+    from hathor.nanocontracts.storage.patricia_trie import Node, NodeId
+
+
+class NodeNCType(NCType['Node']):
+    """ Used internally to (de)serialize a Node into/from the database.
+    """
+
+    __slots__ = ('_key', '_length', '_content', '_children', '_id')
+    _key: NCType[bytes]
+    _length: NCType[int]
+    _content: NCType[bytes | None]
+    _children: NCType[dict[bytes, NodeId]]
+    # XXX: id is not optional, we're indicating that only nodes with id can be stored
+    _id: NCType[NodeId]
```

Comment:
This is currently the only reasons for this class to be "manual", maybe some tweaks to `Node` could simplify this and get rid of this module.